models/eventModel.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+const mongoose = require("mongoose");
+const validator = require("validator");
+
+const eventSchema = new mongoose.Schema({
+    time: {
+        type: Date,
+        default: Date.now,
+        immutable: true
+    }
+}, {discriminatorKey: 'kind'});
+
+const eventConnection = new mongoose.Schema({
+    ip: {
+        type: String,
+        required: true,
+        validate: [validator.isIP, "IP is invalid"],
+        immutable: true
+    },
+    userAgent: {
+        type: String,
+        required: true,
+        immutable: true
+    },
+    user: {
+        type: mongoose.Schema.Types.ObjectId,
```

Comment:
needs ref: ?

Reply:
Do we need to populate ?. I will add them anyway but still something we need to see

Reply:
👍🏼